tree/tree/src/TLeaf.cxx

```diff
@@ -249,7 +249,7 @@ TString TLeaf::GetFullName() const
 TLeaf* TLeaf::GetLeafCounter(Int_t& countval) const
 {
    countval = 1;
-   const char* name = GetTitle();
+   const char *name = GetName();
```

Comment:
```suggestion
   const char *name = GetTitle();
```

Reply:
Ok, but for truncated data types, name = arr[n] where title = d[0,10,10]
so what would be the suggestion instead?

Reply:
I don't see the `name = arr[n]` part:
```
root [0]   TTree t("t", "t");
root [1]   int n;
root [2]   Double32_t arr[64];
root [3]   t.Branch("n", &n);
root [4]   t.Branch("arr", arr, "arr[n]/d[0,0,10]");
root [5]   t.Branch("arr2", arr, "arr2[n]/D");
root [6] auto l = (TLeaf*)t.GetListOfLeaves()->At(2)
(TLeaf *) 0x1bb65a0
root [7] l->GetName()
(const char *) "arr2"
root [8] l->GetTitle()
(const char *) "arr2[n]"
root [9] auto l = (TLeaf*)t.GetListOfLeaves()->At(1)
(TLeaf *) 0x2568a10
root [10] l->GetName()
(const char *) "arr"
root [11] l->GetTitle()
(const char *) "d[0,0,10]"
```
and additionally, `TLeaf::GetCount` seesm to work fine:
```
root [14] l->GetLeafCount()->GetName()
(const char *) "n"
```

Reply:
However it might **not** be working when reading from a file.  So we need to extend the file to add the round trip to and from file.